ICE: enum discriminant depends on generic arguments

Following code ICEs on today's nightly (97e58c0d3 2019-09-20):
```rust
enum Foo {
  A = foo(),
  B = foo(), // <- error: internal compiler error: src/librustc/ty/mod.rs:2400: enum discriminant depends on generic arguments
}

const fn foo<T>() -> isize {
   0
}
```
What's peculiar is that no ICE is shown when the `Foo` enum contains only one item:
```rust
enum Foo {
  A = foo(),
}
```

It's not a regression. Stable shows

```
   Compiling playground v0.0.1 (/playground)
error[E0282]: type annotations needed
 --> src/lib.rs:2:7
  |
2 |   A = foo(),
  |       ^^^ cannot infer type for `T`

error[E0019]: constant contains unimplemented expression type
 --> src/lib.rs:3:7
  |
3 |   B = foo(), // <- error: internal compiler error: src/librustc/ty/mod.rs:2400: enum discriminant depends on generic arguments
  |       ^^^^^

error: internal compiler error: src/librustc/ty/mod.rs:2394: enum discriminant depends on generic arguments
 --> src/lib.rs:3:7
  |
3 |   B = foo(), // <- error: internal compiler error: src/librustc/ty/mod.rs:2400: enum discriminant depends on generic arguments
  |    
```

I don't see how we could be keeping state. Maybe it's happening because there's no "cannot infer type for `T`" error on anything but the first item? So `A` is never evaluated, thus not causing an ICE.

--- a/src/test/ui/consts/issue-64662.rs
+++ b/src/test/ui/consts/issue-64662.rs
@@ -1,0 +1,10 @@
+enum Foo {
+    A = foo(), //~ ERROR: type annotations needed
+    B = foo(), //~ ERROR: type annotations needed
+}
+
+const fn foo<T>() -> isize {
+    0
+}
+
+fn main() {}

--- a/src/test/ui/consts/issue-64662.stderr
+++ b/src/test/ui/consts/issue-64662.stderr
@@ -1,0 +1,15 @@
+error[E0282]: type annotations needed
+  --> $DIR/issue-64662.rs:2:9
+   |
+LL |     A = foo(),
+   |         ^^^ cannot infer type for `T`
+
+error[E0282]: type annotations needed
+  --> $DIR/issue-64662.rs:3:9
+   |
+LL |     B = foo(),
+   |         ^^^ cannot infer type for `T`
+
+error: aborting due to 2 previous errors
+
+For more information about this error, try `rustc --explain E0282`.

